hwmv2: intel_ish: Port to HMWv2

golowanow opened the pull request:
Port `intel_ish` SoCs and board configurations for:
* intel_ish_5_4_1
* intel_ish_5_6_0
* intel_ish_5_8_0

tejlmand reviewed (approved):
lgtm.

tejlmand commented on boards/v2/intel/intel_ish/Kconfig.defconfig:
```diff
@@ -0,0 +1,13 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if TEST
+config TEST_EXTRA_STACK_SIZE
```
minor nit, the construct `if` construct is equivalent to apply `if` on all entries.

So when there is only one entry, then you might just write:
```
config TEST_EXTRA_STACK_SIZE
	int
	default 1024
	depends on TEST
```
(`depends on` is also equivalent to apply `if` on all parts)
Ref: https://www.kernel.org/doc/html/latest/kbuild/kconfig-language.html#

nordicjm commented on boards/v2/intel/intel_ish/Kconfig.intel_ish_5_4_1:
```diff
@@ -0,0 +1,5 @@
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_INTEL_ISH_5_4_1
+  select SOC_INTEL_ISH_5_4_1
```
Kconfigs use tabs

nordicjm commented on boards/v2/intel/intel_ish/Kconfig.intel_ish_5_6_0:
```diff
@@ -0,0 +1,5 @@
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_INTEL_ISH_5_6_0
+  select SOC_INTEL_ISH_5_6_0
```
as above

golowanow replied:
fixed

nordicjm commented on boards/v2/intel/intel_ish/Kconfig.intel_ish_5_8_0:
```diff
@@ -0,0 +1,5 @@
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_INTEL_ISH_5_8_0
+  select SOC_INTEL_ISH_5_8_0
```
as above

golowanow replied:
fixed

nordicjm commented on boards/v2/intel/intel_ish/board.yml:
```diff
@@ -0,0 +1,16 @@
+boards:
+
+  - name: intel_ish_5_4_1
+    vendor: intel
+    socs:
+      - name: intel_ish_5_4_1
+
+  - name: intel_ish_5_6_0
+    vendor: intel
+    socs:
+      - name: intel_ish_5_6_0
+
+  - name: intel_ish_5_8_0
+    vendor: intel
+    socs:
+      - name: intel_ish_5_8_0
```
this should be using revisions

tejlmand replied:
but we don't have revisions for SoCs in Zephyr, only for boards, and those numbers seems to identify the SoC.

I assume somewhat similar to what nRF does with its package variant identification field, like `qfaa`, `ciaa`, etc.

So probably the cleanest approach would be to have a single board, and then identify the soc, something like:
```
board
 name: intel_ish
  vendor: intel
    socs:
      - name: intel_ish_5_4_1
      - name: intel_ish_5_6_0
      - name: intel_ish_5_8_0
```

but I would leave such decision at the discretion of the board maintainer.

nordicjm replied:
We do have board revisions which can select which soc is used, as per https://github.com/zephyrproject-rtos/zephyr/pull/67770/commits/6809aaefb987e5361355b676d9bf87dfe703ca60#diff-e26d2b274c8899adef8f82cb8fcf2a53a96838c4726caa9d7ad11c2afdb886af

golowanow replied:
> but I would leave such decision at the discretion of the board maintainer.

@nashif , @jhedberg, @tbursztyka, @kwd-doodling - what do you think: should we keep ISH for HWMv2 as separate socs/boards for now same as currently, or migrate to board revisions, or  ?

tejlmand replied:
> We do have board revisions which can select which soc is used

I know, I was just pointing to the fact that `5_4_1`, `5_6_0`, `5_8_0` doesn't seem to be a revision of the board, but the revision of the SoC, and therefore using board revisions in this case might be a misnomer as well.

golowanow replied:
> So probably the cleanest approach would be to have a single board, and then identify the soc, something like:
> ...
> but I would leave such decision at the discretion of the board maintainer.

I've changed ish_boards to variants - as a separate (last) commit in this PR.

```
 intel_ish_5_4_1 --> intel_ish/intel_ish_5_4_1
 intel_ish_5_6_0 --> intel_ish/intel_ish_5_6_0
 intel_ish_5_8_0 --> intel_ish/intel_ish_5_8_0
```

@nordicjm, @tejlmand, @nashif , @jhedberg, @tbursztyka, @kwd-doodling - is it what suits v.2 migration and the ISH design ?

nordicjm replied:
Not really no, it should be:
```
intel_ish@5.4.1
intel_ish@5.6.0
intel_ish@5.8.0
```

kwd-doodling replied:
In fact, they're different boards with different SoCs, not one board with different SoCs.
I've tried board revision. unfortunately it's a support of different boards with a same SoC, which doesn't fit our usage.

golowanow replied:
> In fact, they're different boards with different SoCs, not one board with different SoCs. I've tried board revision. unfortunately it's a support of different boards with a same SoC, which doesn't fit our usage.

@kwd-doodling just to double check: does the new `board/soc/variant` naming scheme fit the ISH purposes ?,
i.e. what is proposed with the last commit here to have `intel_ish/intel_ish_5_4_1` etc. board names

tejlmand replied:
> In fact, they're different boards with different SoCs, not one board with different SoCs.

if they are in fact different boards, then I think they should still be modeled as individual boards.
I just had the impression that the `intel_ish` as a board was kind of a phony board, because the ish is a co-processor inside a SoC / SiP.

golowanow replied:
@nashif what do you think about possible optimizations for ISH names ?

golowanow replied:
> @nashif what do you think about possible optimizations for ISH names ?

@nashif, @kwd-doodling - are we going to rename SoC/boards somehow towards hwmv2 in this PR ?
If yes, then please suggest your new naming schema.

nordicjm commented on boards/v2/intel/intel_ish/intel_ish_5_6_0_defconfig:
```diff
@@ -0,0 +1,10 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# uart & console
+CONFIG_CONSOLE=y
+CONFIG_UART_CONSOLE=y
+CONFIG_SERIAL=y
```
Because there are all the same, with a revision, you can have a single defconfig instead of 3

kwd-doodling replied:
before those, definition of SoC is different.
With more features added, they may have more differences later.

nordicjm commented on boards/v2/intel/intel_ish/Kconfig.defconfig:
```diff
@@ -0,0 +1,13 @@
+
+if TEST
+config TEST_EXTRA_STACK_SIZE
+	int
```
`int` should not be here as the Kconfig is created in zephyr itself

golowanow replied:
removed

kwd-doodling commented:
@golowanow Hi Dmitrii, what's HMWv2? why move ISH boards/SoCs into a new v2 sub-directory?

golowanow commented:
> @golowanow Hi Dmitrii, what's HMWv2? why move ISH boards/SoCs into a new v2 sub-directory?

@kwd-doodling more details on HMWv2 are here -  #51831, #67682 or you can contacl me or @nashif on Teams/email.

golowanow committed:
soc: x86: intel_ish: move and convert to HWMv2

Move and convert soc/x86/intel_ish to HWMv2 as soc/intel/intel_ish

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow committed:
boards: x86: intel_ish: move and convert intel_ish boards to HWMv2

Move and convert to HWMv2 the following board configurations:
intel_ish_5_4_1, intel_ish_5_6_0, intel_ish_5_8_0

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow commented:
1) adjusted to #68346 
2) keep intel_ish board configurations as separate boards (no revisions, no variants) [as suggested](https://github.com/zephyrproject-rtos/zephyr/pull/68178#discussion_r1475624554) by @kwd-doodling

golowanow commented:
@kwd-doodling, @tejlmand, @nashif, @57300, @gmarull - up to your attention.

kwd-doodling commented on soc/intel/intel_ish/Kconfig:
```diff
@@ -1,11 +1,10 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
 config SOC_FAMILY_INTEL_ISH
```
move all below "select" into Kconfig.soc with "config SOC_FAMILY_INTEL_ISH".
no need to keep here.

golowanow replied:
> move all below "select" into Kconfig.soc with "config SOC_FAMILY_INTEL_ISH". no need to keep here.

hmm, why you think it needs to move ?

other HWMv2 SoC families have selects in `Kconfig`, e.g. https://github.com/zephyrproject-rtos/zephyr/blob/collab-hwm/soc/gd_gd32/Kconfig

also note this [short explanation](https://github.com/zephyrproject-rtos/zephyr/pull/68078#discussion_r1473052683)

kwd-doodling commented on soc/intel/intel_ish/Kconfig.soc:
```diff
@@ -0,0 +1,13 @@
+
+config SOC_FAMILY
+	default "intel_ish" if SOC_FAMILY_INTEL_ISH
+
```
why having "rsource "*/Kconfig.soc"?
It seems to have */Kconfig.soc included here is better. See other SoC as reference.

golowanow replied:
changed: `Kconfig.soc` moved to `Kconfig`

kwd-doodling commented on soc/intel/intel_ish/Kconfig.defconfig:
why having cache-related config options here removed?

config CACHE_MANAGEMENT
	default y

config X86_MFENCE_INSTRUCTION_SUPPORTED
	default n

config DCACHE_LINE_SIZE
	default 16

and even "select CPU_HAS_DCACHE" for "config SOC_FAMILY_INTEL_ISH".

golowanow replied:
> why having cache-related config options here removed?

same here - it is not yet in the upstream main branch eventually not in the collab-hwm branch

https://github.com/zephyrproject-rtos/zephyr/blob/199487be54246436f0bd75aa9965927db7623f85/soc/x86/intel_ish/Kconfig.defconfig#L1-L30

kwd-doodling commented on soc/intel/intel_ish/intel_ish5/Kconfig.soc:
```diff
@@ -0,0 +1,21 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config SOC_INTEL_ISH_5_4_1
```
previously those 3 have choice wrapped. Why removed?
How to avoid more than one of them are selected?

golowanow replied:
no SoC choice in HWMv2 as its selection is from the chosen board dependency, see also other HWMv2 SoC series, e.g. `soc/gd_gd32/gd32f4xx/Kconfig.soc`

kwd-doodling commented on boards/intel/intel_ish/intel_ish_5_4_1.yaml:
why having below things removed compared with current file in main branch?
  - dma
  - gpio
  - i2c
  - serial
  - spi
  They're not configured in defconfig, but their drivers are ready and can be configured and run.

golowanow replied:
> why having below things removed compared with current file in main branch?
> 
> * dma
> * gpio
> * i2c
> * serial
> * spi
>   They're not configured in defconfig, but their drivers are ready and can be configured and run.

the current upstream main branch doesn't have these, for some reason this extension is not yet upstreamed.
https://github.com/zephyrproject-rtos/zephyr/blob/7551b85f2287d2ae19f9b058688721ac2f04367c/boards/x86/intel_ish/intel_ish_5_4_1.yaml#L1-L14

kwd-doodling commented on boards/intel/intel_ish/intel_ish_5_6_0.yaml:
same comments as to intel_ish_5_4_1.yaml.

golowanow replied:
same thing - it is not yet upstreamed
https://github.com/zephyrproject-rtos/zephyr/blob/7551b85f2287d2ae19f9b058688721ac2f04367c/boards/x86/intel_ish/intel_ish_5_6_0.yaml#L1-L14

kwd-doodling commented on boards/intel/intel_ish/intel_ish_5_8_0.yaml:
same comments as to intel_ish_5_4_1.yaml.

golowanow replied:
same thing - it is not yet upstreamed
https://github.com/zephyrproject-rtos/zephyr/blob/7551b85f2287d2ae19f9b058688721ac2f04367c/boards/x86/intel_ish/intel_ish_5_8_0.yaml#L1-L14

golowanow commented:
* compliance check passing with #68811
* twister fails are not related to this PR

simonguinot commented:
Sorry, I messed up with the reviewers list while removing myself. And some reason I am unable to fix it...

simonguinot commented:
It looks like that at most 15 reviewers can be "manually" selected. Maybe the removal of additional people is due to this limit as well.

golowanow committed:
MAINTAINERS: intel_ish: Adjust to HWMv2 move

Adjust intel/intel_ish SoC and board maintainers to HWMv2 move.

Signed-off-by: Dmitrii Golovanov <dmitrii.golovanov@intel.com>

golowanow commented:
* rebase to the latest changes at `collab-hwm`
* align paths at `MAINTAINERS`
